lib/onix/onix_message.rb

```diff
@@ -104,6 +104,19 @@ def open(arg, force_encoding = nil)
       xml
     end
 
+    def reader(file_path)
+      @reader = Nokogiri::XML::Reader(File.open(file_path, 'r'))
+    end
+
+    def each(&block)
+      @reader.each do |node|
```

Comment:
maybe add 

```ruby
return unless @reader
```
before the iteration to avoid an exception if `reader` has not been called ?